everestrs/everestrs/build.rs

```diff
@@ -0,0 +1,88 @@
+use std::env;
+use std::path::{Path, PathBuf};
+
+struct Libraries {
+    everestrs_sys: PathBuf,
+    framework: PathBuf,
+}
+
+fn find_everest_workspace_root() -> PathBuf {
+    let mut cur_dir =
+        PathBuf::from(env::var("CARGO_MANIFEST_DIR").expect("always set in build.rs execution"));
+
+    // A poor heuristic: We traverse the directories upwards until we find a directory called
+    // everest-framework and hope that this is the EVerest workspace.
+    while cur_dir.parent().is_some() {
+        cur_dir = cur_dir.parent().unwrap().to_path_buf();
+        if cur_dir.join("everest-framework").is_dir() {
+            return cur_dir;
+        }
+    }
+    panic!("everstrs is not build within an EVerest workspace.");
+}
+
+/// Returns the Libraries path if this is a standalone build of everest-framework or None if it is
+/// not.
+fn find_libs_in_everest_framework(root: &Path) -> Option<Libraries> {
+    let everestrs_sys = root.join("everest-framework/build/everestrs/libeverestrs_sys.a");
+    let framework = root.join("everest-framework/build/lib/libframework.so");
+    if everestrs_sys.exists() && framework.exists() {
+        Some(Libraries {
+            everestrs_sys,
+            framework,
+        })
+    } else {
+        None
+    }
+}
+
+/// Returns the Libraries path if this is an EVerest workspace where make install was run in
+/// everest-core/build or None if not.
+fn find_libs_in_everest_core_build_dist(root: &Path) -> Option<Libraries> {
+    let everestrs_sys = root.join("everest-core/build/dist/lib/libeverestrs_sys.a");
+    let framework = root.join("everest-core/build/dist/lib/libframework.so");
+    if everestrs_sys.exists() && framework.exists() {
+        Some(Libraries {
+            everestrs_sys,
+            framework,
+        })
+    } else {
+        None
+    }
+}
+
+/// Takes a path to a library like `libframework.so` and returns the name for the linker, aka
+/// `framework`
+fn libname_from_path(p: &Path) -> String {
+    p.file_stem()
+        .and_then(|os_str| os_str.to_str())
+        .expect("'p' must be valid UTF-8 and have a .so extension.")
+        .strip_prefix("lib")
+        .expect("'p' should start with `lib`")
+        .to_string()
+}
```

Comment:
As mentioned in our call yesterday, I think this is the biggest problem in the current approach: how to marry cargo & cmake. It is not ok to assume that cmake always drives the build process for rust, since this breaks all common development flows for rust engineers. I agree that this approach is brittle and I think we should find a better way to do it, but just assuming that cmake is always running the show is not working.

I suggest we spend time on this, some experimentation and building real modules will hopefully help, but right now I have no clear idea how to go about this.